File: kata-dao/src/main/java/com/socgen/model/Deposit.java
package com.socgen.model;

import jakarta.persistence.*;
import lombok.*;

import java.sql.Date;

@Setter
@Getter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "DEPOSIT")
public class Deposit {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "ID")
    private Long id;

    @Column(name = "AMOUNT")
    private Long amount;

    @Column(name = "DATE")
    @Temporal(TemporalType.DATE)
    private Date date;

    @Column(name = "DESCRIPTION")
    private String description;

}

File: src/app/deposit/Deposit.ts
export class Deposit {

  public id : number | null;
  public amount : number;
  public date : Date;
  public description: string;

  constructor(id: number, amount: number, date: Date, description: string) {
    this.id = id;
    this.amount = amount;
    this.date = date;
    this.description = description;
  }


}
